### PR TITLE
publish: first draft of graph-store migration

### PR DESCRIPTION
**Pull request by liam-fitzgerald**
State adapter to migrate publish to graph-store.

Just want to check that I understand the graph-store API correctly.

Open questions:
- How should titles be set on notes? We can't attach a metadata entry to a particular graph node. Tempted to just parse the metadata out of the text contents (this is what publish already does). Alternatively, we could put a metadata entry into contents and use that? If we opt for parsing the metadata, should we check the validity inside the publish mark?

- I'm leaving the state untouched, so that a) we don't have any irretrievable data loss and b) we can port read indicators forward when notification-store is shipped.

Pending:
- Graph hook implementation

### CODE REVIEW

**Comment by tacryt-socryp** [comment]
Keep in mind that contents within a post is a list, not a singular entry. I think the best thing to do is make contents have two %text entries: the first is the title, the second is the body.

**Comment by liam-fitzgerald** [comment]
> Keep in mind that contents within a post is a list, not a singular entry. I think the best thing to do is make contents of a "publish note" (toplevel node in the graph) have two %text entries: the first is the title, the second is the body.

What about other metadata, like whether comments are enabled?

**Comment by tacryt-socryp** [comment]
I'm okay with this being configured per notebook. perhaps we add an optional key-value map of data of types %boolean, %integer, %date... to metadata-store.

File: pkg/arvo/mar/graph/validator/publish.hoon [hunk 1]
```diff
@@ -9,8 +9,15 @@
...
     =/  ip  ;;(indexed-post p)
-    ?>  (gte 2 (lent index.p.ip))
-    ip
+    |^
+    ?+    index.p.ip  !!
+      [@ @ ~]  ip
+    ::  top level post must have title as first content
+        [@ ~]
+      ?>  ?=(^ contents.p.ip)
+      ?>  ?=(%text -.i.contents.p.ip)
```
Comment (tacryt-socryp) [inline pkg/arvo/mar/graph/validator/publish.hoon:18]: I'm curious: do you think that contents should be more heavily constrained, specifically as [%text %text]~? It is currently not especially constrained: the list can be of any length, and can have any type in the subsequent list items.

Comment (liam-fitzgerald) [reply pkg/arvo/mar/graph/validator/publish.hoon:18]: We have to be able to render the other content types, so I don't think it really makes much of a difference from a development perspective, and it seems unnecessary to limit publish like this. I'd love to see publish as a jupyter style notebook for hoon and I'm sure there will be other usecases that I haven't thought of

**Comment by liam-fitzgerald** [comment]
ready to go into release/graph-publish

**Comment by liam-fitzgerald** [comment]
cc: @tacryt-socryp

**Comment by tacryt-socryp** [comment]
This is obsoleted by #3473.

**Pull request closed without merging**